### PR TITLE
Issue #3201678 by chmez: As a user I want to be able to provide the details about my nationality

### PR DESCRIPTION
**Pull request by ghost**
## Problem
As a user I want to be able to provide the details about my nationality.

## Solution
- add select field for Nationality to social_profile_fields module
  - create new taxonomy “nationality” so CM+ could manage the list
  - reuse the list of countries from [here](https://gist.githubusercontent.com/marijn/274449/raw/0045fb5f54f9ad357e301cf30e23d9834058618a/nationalities.yaml)
  - should be possible to select single value
- should be possible to add the field to registration form
- should be possible for user to hide this information
  - include in social_privacy module

## Issue tracker
- https://www.drupal.org/project/social/issues/3201678
- https://getopensocial.atlassian.net/browse/YANG-5041
- https://getopensocial.atlassian.net/browse/YANG-5357

## How to test
- [x] Enable the "Social Profile Fields" module and log in as SM.
- [x] When you navigate to the "/admin/structure/taxonomy" page then you should see "Nationality" taxonomy vocabulary.
- [x] When you click "Nationality" then you should see a list of taxonomy terms where names are items from [here](https://gist.githubusercontent.com/marijn/274449/raw/0045fb5f54f9ad357e301cf30e23d9834058618a/nationalities.yaml).
- [x] When you log in as a normal member and open the profile edit page then you should see the "Nationality" optional field in the "Phone number and location" section.
- [x] Start typing "ukr" in the "Nationality" field and select the "Ukrainian" item in a drop-down list.
- [x] Put then comma-symbol and start typing "du" and select the "Dutch" item in the drop-down list.
- [x] When you press "Save" then you should see the "Nationality" field with "Ukrainian" and "Dutch" options.

## Release notes
Add the "Nationality" field to the user profile.

### CODE REVIEW

**Review by ribel — APPROVED** [review]
All looks good and works as expected to me.

**Review by ronaldtebrake — CHANGES_REQUESTED** [review]
Hi @OH17L //cc @ribel 

I see you use the countries list as provided by the Address module,
however when you are talking about nationalities, it's not the same as a country.

E.g. I'm from the Netherlands, but my nationality is Dutch.
https://www.vocabulary.cl/Basic/Nationalities.htm

So we can't simply use a list of countries for this unfortunately.

**Comment by ronaldtebrake** [comment]
Hi @OH17L //cc @ribel 
There are some open source lists available we could use as a start https://gist.github.com/marijn/274449 / https://bl.ocks.org/marijn/274449

Hope it helps :) !

File: modules/social_features/social_profile/modules/social_profile_fields/social_profile_fields.install [hunk 1]
```diff
@@ -73,3 +75,43 @@ function _social_profile_fields_update_search_index($index_id = 'social_users')
...
+ * Add a field for nationality.
+ */
+function social_profile_fields_update_8003() {
+  $entity_type_ids = [
+    'taxonomy.vocabulary.' => 'taxonomy_vocabulary',
+    'field.storage.profile.field_profile_' => 'field_storage_config',
+    'field.field.profile.profile.field_profile_' => 'field_config',
+  ];
+
+  $path = drupal_get_path('module', 'social_profile_fields') . '/config/install/';
```
[the remark this answers is not in the document]
Comment (OH17L) [reply modules/social_features/social_profile/modules/social_profile_fields/social_profile_fields.install]: Done.

**Review by ribel — APPROVED** [review]

**Comment by ribel** [comment]
Hi @ronaldtebrake 
@OH17L have replaced countries with actual nationalities as you proposed. 
I have checked it and it looks good to me. You can test it on ECI Demo.
Could you check this as well and tell if we can merge it and include it in release 10.1?

**Comment by ronaldtebrake** [comment]
Hi @ribel @OH17L 
Thank you for the improvements, this looks great! :) 

Would you do me a favor and maybe add a follow-up so we can also export this new field when we export users?
But even without that it can be merged from my side and we can itterate on the export in a later stage. 🚀

**Comment by ribel** [comment]
Thanks, @ronaldtebrake!
I have created follow up here: https://getopensocial.atlassian.net/browse/YANG-5387

**Pull request merged**